File: src/com/l2kt/gameserver/model/actor/stat/DoorStat.java
package com.l2kt.gameserver.model.actor.stat;

import com.l2kt.gameserver.model.L2Skill;
import com.l2kt.gameserver.model.actor.Creature;
import com.l2kt.gameserver.instancemanager.SevenSigns;
import com.l2kt.gameserver.instancemanager.SevenSigns.SealType;
import com.l2kt.gameserver.model.actor.instance.Door;

public class DoorStat extends CreatureStat
{
	private int _upgradeHpRatio;
	
	public DoorStat(Door activeChar)
	{
		super(activeChar);
		
		_upgradeHpRatio = 1;
	}
	
	@Override
	public Door getActiveChar()
	{
		return (Door) super.getActiveChar();
	}
	
	@Override
	public int getMDef(Creature target, L2Skill skill)
	{
		double defense = getActiveChar().getTemplate().getBaseMDef();
		
		switch (SevenSigns.INSTANCE.getSealOwner(SealType.STRIFE))
		{
			case DAWN:
				defense *= 1.2;
				break;
			
			case DUSK:
				defense *= 0.3;
				break;
		}
		
		return (int) defense;
	}
	
	@Override
	public int getPDef(Creature target)
	{
		double defense = getActiveChar().getTemplate().getBasePDef();
		
		switch (SevenSigns.INSTANCE.getSealOwner(SealType.STRIFE))
		{
			case DAWN:
				defense *= 1.2;
				break;
			
			case DUSK:
				defense *= 0.3;
				break;
		}
		
		return (int) defense;
	}
	
	@Override
	public int getMaxHp()
	{
		return super.getMaxHp() * _upgradeHpRatio;
	}
	
	public final void setUpgradeHpRatio(int hpRatio)
	{
		_upgradeHpRatio = hpRatio;
	}
	
	public final int getUpgradeHpRatio()
	{
		return _upgradeHpRatio;
	}
}

File: src/com/l2kt/gameserver/model/actor/stat/DoorStat.kt
package com.l2kt.gameserver.model.actor.stat

import com.l2kt.gameserver.instancemanager.SevenSigns
import com.l2kt.gameserver.instancemanager.SevenSigns.SealType
import com.l2kt.gameserver.model.L2Skill
import com.l2kt.gameserver.model.actor.Creature
import com.l2kt.gameserver.model.actor.instance.Door

class DoorStat(activeChar: Door) : CreatureStat(activeChar) {
    var upgradeHpRatio: Int = 0

    override val maxHp: Int
        get() = super.maxHp * upgradeHpRatio

    init {

        upgradeHpRatio = 1
    }

    override val activeChar: Door? get() = super.activeChar as Door?

    override fun getMDef(target: Creature?, skill: L2Skill?): Int {
        var defense = activeChar!!.template.baseMDef

        when (SevenSigns.getSealOwner(SealType.STRIFE)) {
            SevenSigns.CabalType.DAWN -> defense *= 1.2

            SevenSigns.CabalType.DUSK -> defense *= 0.3
        }

        return defense.toInt()
    }

    override fun getPDef(target: Creature?): Int {
        var defense = activeChar!!.template.basePDef

        when (SevenSigns.getSealOwner(SealType.STRIFE)) {
            SevenSigns.CabalType.DAWN -> defense *= 1.2

            SevenSigns.CabalType.DUSK -> defense *= 0.3
        }

        return defense.toInt()
    }
}